apps/sms/services/js/moz_mobile_connections/moz_mobile_connections_shim.js

```diff
@@ -0,0 +1,106 @@
+/* global BridgeServiceMixin,
+          BroadcastChannel,
+          MozSettingsShim
+*/
+
+/* exported MozMobileConnectionsShim */
+
+(function(exports) {
+'use strict';
+
+/**
+ * Name of the service for mozMobileConnections API shim.
+ * @type {string}
+ */
+const SERVICE_NAME = 'moz-mobile-connections-shim';
+
+/**
+ * Array of method names that need to be exposed for API shim.
+ * @type {Array.<string>}
+ */
+const METHODS = Object.freeze(['getServiceIdByIccId', 'switchMmsSimHandler']);
+
+var mozMobileConnections = null;
+
+function setMmsSimServiceId(id) {
+  // DSDS: mms & data are both necessary for connection switch.
+  MozSettingsShim.set({
+    'ril.mms.defaultServiceId': id,
+    'ril.data.defaultServiceId': id
+  });
+}
+
+var MozMobileConnectionsShim = {
+  init(appInstanceId, conns) {
+    if (!conns) {
+      console.error('Invalid mozMobileConnections for shim initialization');
+      return;
+    }
+
+    var broadcastChannelName = `${SERVICE_NAME}-channel-${appInstanceId}`;
+    mozMobileConnections = conns;
+
+    this.initService(
+      new BroadcastChannel(broadcastChannelName)
+    );
+  },
+
+  /* Methods */
+
+  /**
+   * Shim for returning the service id that matches the icc id.
+   * @returns {?number} matched service id or null if unmatched.
+   */
+  getServiceIdByIccId(iccId) {
+    var index = Array.prototype.findIndex.call(mozMobileConnections, (conn) =>
+      iccId === conn.iccId
+    );
+
+    return index > -1 ? index : null;
+  },
+
+  /**
+   * Switching the default mms/data service id by icc id.
+   * @param {number} iccId Id that could map to the target SIM.
+   * @returns {Promise} Promise that returned when sim switching is ready or
+   *  failed.
+   */
+  switchMmsSimHandler(iccId) {
```

Comment:
I'm thinking of removing the entire settings.js  when it's totally replicable by settings shim and other client/service. Maybe you have some concern about the memory consumption? If we can't split the settings.js into NGA for 2.5 , then we might need to remove the duplicate methods, wdyt?